### PR TITLE
Mathematics for Machine Learning (ML)

### LINKED ISSUE #651: Mathematics for Machine Learning (ML)
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

Mathematics is everywhere. It has helped us achieve a lot through our history.  From finance to time management, the use of mathematics has been very essential. In this article, we will be learning the importance of mathematics in Machine Learning (ML). 

### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

By the end of this article, the reader should be able to understand the fields necessary to master Machine Learning (ML). They include:

1. Linear Algebra
2. Multivariate Calculus
3. Probability and Statistics
4. Algorithms

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

### PR DESCRIPTION
**Pull request by Bayler**
Fixed #651

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/mathematics-for-machine-learning/index.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+7. Chain Rule.
+8. Vector-Values Functions
+9. Directional Gradient
+
+#### Some Applications of Calculus in ML
+
+1. Optimization techniques such as the Gradient Descent, Adam, RMSProp, and Ada Delta methods use Calculus to help find the local and global minimum and maxima.
+2. Backpropagation is the main algorithm used to train neural networks. It is achieved by using calculus, which utilizes concepts such as chain rule and partial derivatives.
+
+### Probability $ Statistics
```
Comment (lalith1403) [inline articles/mathematics-for-machine-learning/index.md:74]: Probability and Statistics

File: articles/mathematics-for-machine-learning/index.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+4. Algorithms (15.0%)
+5. Others (10.0%)
+
+This article is not intended to go into details about the mathematical calculations in the various topics. Instead, it's meant to give you a general overview of the math topics behind ML and how they are used. 
+
+Let's dive into it.
+
+### Linear Algebra
+
+Linear Algebra covers a significant part of the mathematical concepts used for Machine Learning. According to [Wikipedia](https://en.wikipedia.org/wiki/Linear_algebra/), Linear Algebra is a mathematics domain concerned with linear equations and the way they are represented in vector spaces and through matrices.
```
Comment (lalith1403) [inline articles/mathematics-for-machine-learning/index.md:31]: Just mention Linear algebra is a subdomain dealing with linear systems of equations. Remove the line according to wikipedia.

File: articles/mathematics-for-machine-learning/index.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+The topics in Linear Algebra essential for understanding the methods used in machine learning include:
```
Comment (lalith1403) [inline articles/mathematics-for-machine-learning/index.md:33]: Add links to resources to each of these topics. See this as a list of resources. So adding links to each of these resources will enhance the value delivered. Give a link for each subtopic under each subject.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Add links to all the topics mentioned. Find resources and insert links. Let us make sure anyone starting with ML can use this as reference material to access top-notch material.

**Comment by hectorkambow** [comment]
Plagiarism check done

**Pull request merged**